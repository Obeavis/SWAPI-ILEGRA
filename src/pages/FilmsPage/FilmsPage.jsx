import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { withRouter } from "react-router-dom";
import { Container, Row } from "react-bootstrap";
import Films from "../../containers/Films/Films";
import SomethingWentWrong from "../SomethingWentWrong/SomethingWentWrong"
import BB8Loader from "../../components/BB8Loader/BB8Loader";

const FilmsPage = () => {
	const dispatch = useDispatch();

	useEffect(() => {
		dispatch({ type: 'GET_ALL_FILMS' })
		// eslint-disable-next-line
	}, []);

	const allFilms = useSelector(state => state.allFilms);
	return (
		<div className="films">
			{allFilms.error === true ?
				<SomethingWentWrong />

				: allFilms.films === null ?
					<Container >
						<Row className="justify-content-center mt-5">
							<BB8Loader />
						</Row>
					</Container>
					: <Films films={allFilms.films} />
			}
		</div>
	)
}

export default withRouter(FilmsPage);