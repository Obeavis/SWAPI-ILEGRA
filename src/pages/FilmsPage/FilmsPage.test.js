import React from "react";
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom';
import configureStore from "redux-mock-store";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import FilmsPage from "./FilmsPage";

configure({ adapter: new Adapter() });



describe("FilmsPage", () => {
    const mockStore = configureStore();
    let store;

    it("Matches the snapshot", () => {
        store = mockStore({
            allFilms: {
                films: [
                    { "episodeID": 4, "title": "A New Hope", "releaseDate": "1977 - 05 - 25" },
                    { "episodeID": 5, "title": "The Empire Strikes Back", "releaseDate": "1980 - 05 - 17" },
                    { "episodeID": 6, "title": "Return of the Jedi", "releaseDate": "1983 - 05 - 25" },
                    { "episodeID": 1, "title": "The Phantom Menace", "releaseDate": "1999 - 05 - 19" },
                    { "episodeID": 2, "title": "Attack of the Clones", "releaseDate": "2002 - 05 - 16" },
                    { "episodeID": 3, "title": "Revenge of the Sith", "releaseDate": "2005 - 05 - 19" },
                    { "episodeID": 7, "title": "The Force Awakens", "releaseDate": "2015 - 12 - 11" },
                ],
                error: false
            }
        });

        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <FilmsPage />
                </MemoryRouter>
            </Provider>
        );

        const tree = renderer.create(wrapper).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it("Renders the <FilmsPage /> with 7 films", () => {
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <FilmsPage />
                </MemoryRouter>
            </Provider>
        );
        const quantity = wrapper.find('Film');

        expect(quantity).toHaveLength(7);
    });

    it("Renders <BB8Loader />", () => {
        store = mockStore({
            allFilms: {
                films: {},
                error: false
            }
        });

        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <FilmsPage />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.find('BB8Loader').exists()).toEqual(true)
    });

    it("Renders <SomethingWentWrong />", () => {
        store = mockStore({
            allFilms: {
                films: {},
                error: true
            }
        });

        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <FilmsPage />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.find('SomethingWentWrong').exists()).toEqual(true)
    });

})