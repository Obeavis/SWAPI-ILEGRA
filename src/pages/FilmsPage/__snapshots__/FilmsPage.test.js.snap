// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsPage Matches the snapshot 1`] = `
<div
  className="films"
>
  <div
    className="film-select container"
  >
    <div
      className="row"
    >
      <div
        className="mb-5 col-xl-4 col-lg-6"
        onClick={[Function]}
      >
        <div
          className="box-films box"
        >
          <img
            alt="A New Hope"
            className="img-fluid"
            src="episode-4.jpg"
          />
          <span
            className="span-box"
          >
            <span
              className="thumb-title text-bold text-center"
            >
              A New Hope
            </span>
            <span
              className="thumb-date text-1 text-bold"
            >
              <time
                dateTime={null}
              >
                Invalid date
              </time>
            </span>
          </span>
        </div>
      </div>
      <div
        className="mb-5 col-xl-4 col-lg-6"
        onClick={[Function]}
      >
        <div
          className="box-films box"
        >
          <img
            alt="The Empire Strikes Back"
            className="img-fluid"
            src="episode-5.jpg"
          />
          <span
            className="span-box"
          >
            <span
              className="thumb-title text-bold text-center"
            >
              The Empire Strikes Back
            </span>
            <span
              className="thumb-date text-1 text-bold"
            >
              <time
                dateTime={null}
              >
                Invalid date
              </time>
            </span>
          </span>
        </div>
      </div>
      <div
        className="mb-5 col-xl-4 col-lg-6"
        onClick={[Function]}
      >
        <div
          className="box-films box"
        >
          <img
            alt="Return of the Jedi"
            className="img-fluid"
            src="episode-6.jpg"
          />
          <span
            className="span-box"
          >
            <span
              className="thumb-title text-bold text-center"
            >
              Return of the Jedi
            </span>
            <span
              className="thumb-date text-1 text-bold"
            >
              <time
                dateTime={null}
              >
                Invalid date
              </time>
            </span>
          </span>
        </div>
      </div>
      <div
        className="mb-5 col-xl-4 col-lg-6"
        onClick={[Function]}
      >
        <div
          className="box-films box"
        >
          <img
            alt="The Phantom Menace"
            className="img-fluid"
            src="episode-1.jpg"
          />
          <span
            className="span-box"
          >
            <span
              className="thumb-title text-bold text-center"
            >
              The Phantom Menace
            </span>
            <span
              className="thumb-date text-1 text-bold"
            >
              <time
                dateTime={null}
              >
                Invalid date
              </time>
            </span>
          </span>
        </div>
      </div>
      <div
        className="mb-5 col-xl-4 col-lg-6"
        onClick={[Function]}
      >
        <div
          className="box-films box"
        >
          <img
            alt="Attack of the Clones"
            className="img-fluid"
            src="episode-2.jpg"
          />
          <span
            className="span-box"
          >
            <span
              className="thumb-title text-bold text-center"
            >
              Attack of the Clones
            </span>
            <span
              className="thumb-date text-1 text-bold"
            >
              <time
                dateTime={null}
              >
                Invalid date
              </time>
            </span>
          </span>
        </div>
      </div>
      <div
        className="mb-5 col-xl-4 col-lg-6"
        onClick={[Function]}
      >
        <div
          className="box-films box"
        >
          <img
            alt="Revenge of the Sith"
            className="img-fluid"
            src="episode-3.jpg"
          />
          <span
            className="span-box"
          >
            <span
              className="thumb-title text-bold text-center"
            >
              Revenge of the Sith
            </span>
            <span
              className="thumb-date text-1 text-bold"
            >
              <time
                dateTime={null}
              >
                Invalid date
              </time>
            </span>
          </span>
        </div>
      </div>
      <div
        className="mb-5 col-xl-4 col-lg-6"
        onClick={[Function]}
      >
        <div
          className="box-films box"
        >
          <img
            alt="The Force Awakens"
            className="img-fluid"
            src="episode-7.jpg"
          />
          <span
            className="span-box"
          >
            <span
              className="thumb-title text-bold text-center"
            >
              The Force Awakens
            </span>
            <span
              className="thumb-date text-1 text-bold"
            >
              <time
                dateTime={null}
              >
                Invalid date
              </time>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
